File: z-clients/javascript/src/models/metricType.ts
// this file is @generated

export enum MetricType {
    Counter = 'counter',
    Gauge = 'gauge',
    }

export const MetricTypeSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): MetricType {
        return object;
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: MetricType): any {
        return self;
    }
}

File: z-clients/rust/src/models/metric_type.rs
// this file is @generated
use std::fmt;

use serde::{Deserialize, Serialize};

#[derive(Clone, Copy, Debug, Serialize, Deserialize)]
pub enum MetricType {
    #[serde(rename = "counter")]
    Counter,
    #[serde(rename = "gauge")]
    Gauge,
}

impl fmt::Display for MetricType {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        let value = match self {
            Self::Counter => "counter",
            Self::Gauge => "gauge",
        };
        f.write_str(value)
    }
}
